### DISCUSSION TITLE
TGCF is not forwading any message! (TGCF Web)

### DISCUSSION THREAD

**Discussion by iamyashg** [aahnik/tgcf#463] [Q&A] [answered]
I have deployed tgcf on replit (it's always on as I have hacker plan). 

Its not forwading any message from source to destination

Below are the logs :

                    INFO     Loaded plugin replace     __init__.py:98
                    INFO     using session string       config.py:212
                    INFO     Connecting to       mtprotosender.py:226
                             149.154.167.51:443/                     
                             TcpFull...                              
                    INFO     Connection to       mtprotosender.py:277
                             149.154.167.51:443/                     
                             TcpFull complete!                       
                    INFO     The server closed the  connection.py:335
                             connection                              
                    INFO     Connection closed   mtprotosender.py:507
                             while receiving                         
                             data                                    
                    INFO     Closing current     mtprotosender.py:350
                             connection to begin                     
                             reconnect...                            
                    INFO     Connecting to       mtprotosender.py:226
                             149.154.167.51:443/                     
                             TcpFull...                              
                    INFO     Connection to       mtprotosender.py:277
                             149.154.167.51:443/                     
                             TcpFull complete!

**Comment by 25gabriel-cmyk** [comment]
Have you try with another TG account?
What you use for TG Login? Bot Token or User session string?
Also what is your source-destination?
Channel to Channel, Channel to Group, or Group to Channel?

**Comment by iamyashg** [comment]
> Have you try with another TG account?
> What you use for TG Login? Bot Token or User session string?
> Also what is your source-destination?
> Channel to Channel, Channel to Group, or Group to Channel?
> 

1. No, I have not tried with other account, but the account from which I am trying had run TGCF for almost a year so I don't think so that problem is with account.

2. I have used user session string. But I have also tried with bot token, it's still not working. Dunno why?

3. Source Destination is private channel made by me only for testing purpose.

4. It's channel to channel, I have tried.


Btw in your replit is it forwarding? If yes then from where did you get your session string.

**Reply by 25gabriel-cmyk** [reply]
- Thats all the logs you got? if you click "load more logs" what you get? I guess your app didn't successful create connection with TG server.

- Go check on your TG account, Setting--Privacy and Security--Device.  See if there's active session logged in from same IP address with your TGCF hosted.
![IMG_20230704_042001](https://github.com/aahnik/tgcf/assets/134001939/7ccfe271-25a3-4cba-b933-e433bda08f84)

- Try start from beginning for your TG account : 
   >> Terminate all other session except from your phone. 
  >> Delete your old telegram app. Create new app so you get new API_ID and API_HASH.
  >> Generate new session string for your TG account

**Reply by iamyashg** [reply]
Yes, its making session but not working..

![image](https://github.com/aahnik/tgcf/assets/72647281/c8fb274c-171c-4df5-89f2-b9f5ba41ba3e)

Logs :

tgcf
Running latest tgcf version 1.1.8


```
[07/01/23 12:48:37] INFO     Verbosity turned on! This is   cli.py:57
                             suitable for debugging                  
                    INFO     Failed to load SSL library: libssl.py:79
                             <class 'OSError'> (no                   
                             library called "ssl" found)             
                    INFO     cryptg detected, it will be    aes.py:19
                             used for encryption                     
                    INFO     tgcf.config.json detected! config.py:109
                    WARNING  You have not set a         config.py:254
                             password to protect the                 
                             web access to tgcf.                     
                             The default password                    
                             `tgcf` is used.                         
                    INFO     config.py got executed     config.py:259
[07/01/23 12:48:38] INFO     First party plugin fmt    __init__.py:82
                             loaded!                                 
                    INFO     check=True                     fmt.py:16
                             style=<Style.PRESERVE:                  
                             'preserve'>                             
                    INFO     Loaded plugin fmt         __init__.py:98
                    INFO     First party plugin        __init__.py:82
                             replace loaded!                         
                    INFO     check=True                 replace.py:16
                             text={'[.\\s]*[.\\s]*[.\\s              
                             ]*[.\\s]*[.\\s]*[.\\s]*[.\              
                             \s]*[.\\s]*[.\\s]*[.\\s]*[              
                             .\\s]*[.\\s]*[.\\s]*':                             
                             regex=True                              
                    INFO     Loaded plugin replace     __init__.py:98
                    INFO     using session string       config.py:212
                    INFO     Connecting to       mtprotosender.py:226
                             149.154.167.51:443/                     
                             TcpFull...                              
                    INFO     Connection to       mtprotosender.py:277
                             149.154.167.51:443/                     
                             TcpFull complete!                       
                    INFO     The server closed the  connection.py:335
                             connection                              
                    INFO     Connection closed   mtprotosender.py:507
                             while receiving                         
                             data                                    
                    INFO     Closing current     mtprotosender.py:350
                             connection to begin                     
                             reconnect...                            
                    INFO     Connecting to       mtprotosender.py:226
                             149.154.167.51:443/                     
                             TcpFull...                              
                    INFO     Connection to       mtprotosender.py:277
                             149.154.167.51:443/                     
                             TcpFull complete!
```

**Reply by 25gabriel-cmyk** [reply]
•  Iam pretty sure the problem on your TG account mate. Currently on TG jail maybe?
•  Have you try to forward message manually from source to destination? No limitations to send message to certain group?  that account can start direct message to non contact ?
• Most of all, just like I said before, create new App and Hash ID by that account. GENERATE NEW USER SESSION.
Iam using this to generate user session https://github.com/aahnik/tg-login

Also i will try to use TGCF i deployed on replit, will let you know the result

**Comment by iamyashg** [comment]
manually it works, and i have no limitation i have checked in spambot 

in my case https://github.com/aahnik/tg-login this is not working its get stucked so using @SessionStringGeneratorZBot Telegram bot

**Reply by 25gabriel-cmyk** [reply]
I ran TGCF deployed on Replit, got same log and error like you (the messages were not forwarded)

Actually TGCF has never successfully established a session on your telegram.
The session you see on telegram is the session created by SessionStringGeneratorZBot when you generate user session.
![IMG_20230705_204029_134](https://github.com/aahnik/tgcf/assets/134001939/143f3f1d-fb9d-4753-b97e-dd77b739d64f)

I think the issue is on this :
Failed to load SSL library: libssl.py:79
                             <class 'OSError'> (no                   
                             library called "ssl" found)

You can try to reach Replit support or their community forum mate.

**Comment by iamyashg** [comment]
but old version of tgcf was working on replit when i was using it in past which have CLI UI.. so now this issue can only be fix by @aahnik as he is the creator of tgcf and also he know using replit too, so making some modification with session part can help us to run tgcf on replit successfuly

**Reply by 25gabriel-cmyk** [reply]
I made some change on replit.nix file, something look like this 

```
{ pkgs }: {
  deps = [
    pkgs.python310Full
    pkgs.replitPackages.prybar-python310
    pkgs.replitPackages.stderred
    pkgs.openssl_1_1
  ];
  env = {
    PYTHON_LD_LIBRARY_PATH = pkgs.lib.makeLibraryPath [
      # Needed for pandas / numpy
      pkgs.stdenv.cc.cc.lib
      pkgs.zlib
      # Needed for pygame
      pkgs.glib
      # Needed for matplotlib
      pkgs.xorg.libX11
      # Needed for OpenSSL
      pkgs.openssl_1_1
    ];
    PYTHONHOME = "${pkgs.python310Full}";
    PYTHONBIN = "${pkgs.python310Full}/bin/python3.10";
    LANG = "en_US.UTF-8";
    STDERREDBIN = "${pkgs.replitPackages.stderred}/bin/stderred";
    PRYBAR_PYTHON_BIN = "${pkgs.replitPackages.prybar-python310}/bin/prybar-python310";
  };
}

```
the SSL's library issue fixed but the messages didn't forwarded. Can't dig more as no more logs generated after that.
Hope @aahnik have time to solve this

**Comment by iamyashg** [comment, marked as answer]
**Solved the issue!** 

Solution and Cause: _https://github.com/aahnik/tgcf/issues/469#issuecomment-1627672189_
